### PR TITLE
Development

### PR DESCRIPTION
**Pull request by gedfaj**

### CODE REVIEW

File: models/books.js [hunk 1]
```diff
@@ -35,6 +35,13 @@ const BookSchema = new mongoose.Schema({
...
+    post: true,
+    bookId: this._id,
+  };
+};
 
-module.exports = { Book }
\ No newline at end of file
+const Book = mongoose.model('books', BookSchema);
+
+module.exports = { Book };
```
Comment (Asetre) [inline models/books.js]: don't need to export it as an object
module.exports = Book

File: routes/books.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    Book.findById(id, (err, doc) => {
+        if (err) {
+          return res.status(400).send(err);
+        }
+        res.send(doc)
+    })
+});
+
+router.get('/books', (req, res) => {
```
Comment (Asetre) [inline routes/books.js]: destructure
const {skip, limit....} = req.query

Comment (Asetre) [reply routes/books.js]: also use const if data will not change

File: routes/books.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  Book.find()
+    .skip(skip)
+    .sort({_id:order})
+    .limit(limit)
+    .exec((err, docs) => {
+      if(err) {
+        return res.status(400).send(err);
+      }
+      res.send(docs)
+    })
```
Comment (Asetre) [inline routes/books.js]: catch error and send a response or else response will hang

File: routes/books.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  })
+  .then(doc => res.status(201).send(doc.bookPosted()))
+  .catch(err => {
+    console.error(err);
+    res.status(500).send({message: 'Internal server error'})
+  });
+});
+
+  router.put('/book/:id', (req, res) => {
+    if (!(req.params.id && req.body.id && req.params.id === req.body.id)) {
```
Comment (Asetre) [inline routes/books.js]: duplicate req.body.id

File: server.js [hunk 5]
```diff
@@ -2,18 +2,26 @@
...
 
 mongoose.Promise = global.Promise;
 
+// middleware
 app.use(bodyParser.urlencoded({extended: false}));
 app.use(bodyParser.json());
 app.use(cookieParser());
 
+// handle routes
+app.use('/users/api', users);
```
Comment (Asetre) [inline server.js]: why is there /api ?
in the real world you would just use a sub domain
api.yoursite.com/user
api.yoursite.com/books

Comment (Asetre) [reply server.js]: front end runs on yoursite.com
back end runs on api.yoursite.com

File: routes/books.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    review,
+    pages,
+    rating,
+    price,
+    reviewerId
+  })
+  .then(doc => res.status(201).send(doc.bookPosted()))
+  .catch(err => {
+    console.error(err);
+    res.status(500).send({message: 'Internal server error'})
```
[the remark this answers is not in the document]
Comment (Asetre) [reply routes/books.js]: also always return when you send back a response

**Comment by gedfaj** [comment]
Please review changes

**Pull request merged**